plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,224 @@
+                .map(column -> RowType.field(column.getBaseColumnName(), column.getType()))
+                .collect(toImmutableList());
+
+        ImmutableList.Builder<ColumnMetadata> columnMetadataBuilder = ImmutableList.builder();
+
+        if (!this.partitionFields.isEmpty()) {
+            columnMetadataBuilder.add(new ColumnMetadata("partition", RowType.from(this.partitionFields)));
+        }
+
+        columnMetadataBuilder.add(new ColumnMetadata("file_count", BIGINT));
+        columnMetadataBuilder.add(new ColumnMetadata("total_size", BIGINT));
+
+        this.tableMetadata = new ConnectorTableMetadata(tableName, columnMetadataBuilder.build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        if (partitionColumns.isEmpty()) {
+            return new EmptyPageSource();
+        }
+
+        return new FixedPageSource(buildPages(session));
+    }
+
+    private List<Page> buildPages(ConnectorSession session)
+    {
+        PageListBuilder pageListBuilder = PageListBuilder.forTable(tableMetadata);
+        MetadataEntry metadataEntry = transactionLogAccess.getMetadataEntry(session, this.tableSnapshot);
+        ProtocolEntry protocolEntry = transactionLogAccess.getProtocolEntry(session, this.tableSnapshot);
+        ColumnMappingMode columnMappingMode = getColumnMappingMode(metadataEntry, protocolEntry);
+
+        try (Stream<AddFileEntry> activeFiles = transactionLogAccess.loadActiveFiles(session, this.tableSnapshot, metadataEntry, protocolEntry, TupleDomain.all(), alwaysTrue())) {
+            for (Map.Entry<Map<String, Optional<String>>, DeltaLakePartitionStatistics> partitionEntry : getStatisticsByPartition(activeFiles).entrySet()) {
+                Map<String, Optional<String>> partitionValue = partitionEntry.getKey();
+                DeltaLakePartitionStatistics deltaLakePartitionStatistics = partitionEntry.getValue();
+
+                RowType partitionValuesRowType = RowType.from(this.partitionFields);
+                SqlRow partitionValuesRow = buildRowValue(partitionValuesRowType, fields -> {
+                    for (int i = 0; i < partitionColumns.size(); i++) {
+                        DeltaLakeColumnHandle column = partitionColumns.get(i);
+                        Type type = column.getType();
+                        Optional<String> value = switch (columnMappingMode) {
+                            case NONE:
+                                yield partitionValue.get(column.getBaseColumnName());
+                            case ID, NAME:
+                                yield partitionValue.get(column.getBasePhysicalColumnName());
```

Comment:
When does getBaseColumnName return different value from getBasePhysicalColumnName in case of NONE?